server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoConnectionService.java

```diff
@@ -835,6 +849,204 @@ public Response validateConnectionName( final @Context HttpHeaders headers,
+     * @throws KomodoRestException
+     *             if there is a problem deploying the vdb
+     */
+    @POST
+    @Path( V1Constants.CONNECTION_DEPLOY_VDB_SEGMENT + StringConstants.FORWARD_SLASH + V1Constants.CONNECTION_PLACEHOLDER )
+    @Produces( { MediaType.APPLICATION_JSON } )
+    @ApiOperation( value = "Deploy a VDB for the workspace connection" )
+    @ApiResponses(value = {
+            @ApiResponse(code = 406, message = "Only JSON is returned by this operation"),
+            @ApiResponse(code = 403, message = "An error has occurred.")
+        })
+    public Response deployVdbForConnection( final @Context HttpHeaders headers,
+                                     final @Context UriInfo uriInfo,
+                                     @ApiParam( value = "Name of the connection", required = true )
+                                     final @PathParam( "connectionName" ) String connectionName ) throws KomodoRestException {
+
+        SecurityPrincipal principal = checkSecurityContext(headers);
+        if (principal.hasErrorResponse())
+            return principal.getErrorResponse();
+
+        List<MediaType> mediaTypes = headers.getAcceptableMediaTypes();
+        if (! isAcceptable(mediaTypes, MediaType.APPLICATION_JSON_TYPE))
+            return notAcceptableMediaTypesBuilder().build();
+
+        // Error if the connection name is missing
+        if (StringUtils.isBlank( connectionName )) {
+            return createErrorResponseWithForbidden(mediaTypes, RelationalMessages.Error.CONNECTION_SERVICE_MISSING_NAME);
+        }
+
+        UnitOfWork uow = null;
+
+        try {
+            uow = createTransaction( principal, "deployVdbForConnection", false ); //$NON-NLS-1$
+
+            // Get the connection
+            Connection connection = findConnection(uow, connectionName);
+            if (connection == null)
+                return commitNoConnectionFound(uow, mediaTypes, connectionName);
+            String jndiName = connection.getJndiName(uow);
+            String driverName = connection.getDriverName(uow);
+            
+            // Name of VDB to be created, based on connection
+            String vdbName = connectionName + CONNECTION_VDB_SUFFIX;
+            
+            // If VDB with name already exists, delete it
+            Repository repo = this.kengine.getDefaultRepository();
+            final WorkspaceManager mgr = WorkspaceManager.getInstance( repo, uow );
+            String repoPath = repo.komodoWorkspace( uow ).getAbsolutePath();
+            
+            final Vdb existingVdb = findVdb( uow, vdbName );
+
+            if ( existingVdb != null ) {
+                mgr.delete(uow, existingVdb);
+            }
+            
+            // Create new VDB
+            String vdbPath = repoPath + "/" + vdbName;
+            final Vdb vdb = getWorkspaceManager(uow).createVdb( uow, null, vdbName, vdbPath );
+            vdb.setDescription(uow, "Vdb for connection "+connectionName);
```

Comment:
This code should be shared with KomodoVdbService so that both are doing the same job. Otherwise, such duplication will produce greater effort to maintain.